File: examples_ios/MyApp/Sources/SurveyLayouts/SimpleCardLayout/Items/SimpleCardMultiItem.swift
import Foundation
import MobileSdk
import UIKit

class SimpleCardMultiItem: SimpleCardItem {
    @IBOutlet var stackView: UIStackView!
    @IBOutlet var cstrStackHeight: NSLayoutConstraint!
    
    private var question: MultiQuestion!
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }
    
    func setupQuestion(question: MultiQuestion) {
        self.question = question
        lblHeader.text = question.text.get()
        
        createAnswers(question: question)
        
        handleValidation(errors: question.errors)
    }
    
    private func createAnswers(question: MultiQuestion) {
        for answer in question.answers {
            let selectItem: SimpleCardSelectItem = Utils.fromNib()
            selectItem.setupItem(checkbox: true, answer: answer, selected: question.get(answer: answer))
            selectItem.delegate = self
            stackView.addArrangedSubview(selectItem)
        }
        cstrStackHeight.constant = CGFloat(question.answers.count) * 36
    }
    
    override func onNext() -> Bool {
        return validate(question: question)
    }
    
    override func onBack() -> Bool {
        return true
    }
}

extension SimpleCardMultiItem: SimpleCardSelectItemDelegate {
    func onRadioClicked(answer: QuestionAnswer) {
        // Nothing to do
    }
    
    func onCheckClicked(answer: QuestionAnswer, selected: Bool) {
        question.set(answer: answer, select: selected)
    }
}

File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/simpleCardLayout/items/SimpleCardMultiItem.kt
package com.confirmit.testsdkapp.layouts.simplecardlayout.items

import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import com.confirmit.mobilesdk.ui.questions.MultiQuestion
import com.confirmit.mobilesdk.ui.questions.QuestionAnswer
import com.confirmit.testsdkapp.R

class SimpleCardMultiItem(view: View) : SimpleCardItem(view), SimpleCardSelectedItemClickedListener {

    private lateinit var question: MultiQuestion

    private val listView: LinearLayout = view.findViewById(R.id.ctnList)
    private val itemList: MutableList<SimpleCardSelectedItem> = mutableListOf()

    fun setupQuestion(question: MultiQuestion) {
        this.question = question
        txtHeader.text = question.text.get()

        createAnswers(question)

        handleValidation(question.errors)
    }

    private fun createAnswers(question: MultiQuestion) {
        listView.removeAllViews()
        itemList.clear()
        question.answers.forEach {
            val view = LayoutInflater.from(getView().context).inflate(R.layout.simple_card_select_item, listView, false)
            val selectItem = SimpleCardSelectedItem(view)
            selectItem.setupItem(true, it, question.get(it))
            selectItem.listener = this
            listView.addView(view)
            itemList.add(selectItem)
        }
    }

    override fun onNext(): Boolean {
        return validate(question)
    }

    override fun onBack(): Boolean {
        return true
    }

    override fun onRadioClicked(answer: QuestionAnswer) {
    }

    override fun onCheckClicked(answer: QuestionAnswer, selected: Boolean) {
        question.set(answer, selected)
    }
}